lib/srfi/26/expand-cut.scm

```diff
@@ -0,0 +1,46 @@
+(define (expand-cut id close? list<argument-or-slot>)
+
+  (define (gs s) (gensym (string->uninterned-symbol s)))
+
+  (let cons-closure ((l<argument-or-slot> list<argument-or-slot>)
+                     (let-over-lambda-bindings '())
+                     (formals '())
+                     (procedure-application '()))
+
+    (if (null? l<argument-or-slot>)
+
+        `(##let ,let-over-lambda-bindings
+           (##lambda ,(reverse formals)
+             ,(reverse procedure-application)))
+
+        (let ((argument-or-slot (car l<argument-or-slot>)))
+
+          (case argument-or-slot
+
+            ((<>)
+             (let ((parameter (gs "slot")))
+               (cons-closure (cdr l<argument-or-slot>)
+                             let-over-lambda-bindings
+                             (cons parameter formals)
+                             (cons parameter procedure-application))))
+
+            ((<...>)
+             (if (null? (cdr l<argument-or-slot>))
+                 (let ((rest-parameter (gs "rest-slot")))
+                   `(##let ,let-over-lambda-bindings
+                      (##lambda (,@(reverse formals) . ,rest-parameter)
+                        (##apply ,@(reverse procedure-application) ,rest-parameter))))
+                 (error "<...> in non-tail position:" (cons id list<argument-or-slot>))))
```

Comment:
Using `error` is not ideal because the error message will not contain location information.  It is better to use a call to `##raise-expression-parsing-exception` (but this will require using `##define-syntax` instead of `define-macro`).  An alternative is to generate the call

    `(syntax-error "<...> in non-tail position:" ,@(cons id list<argument-or-slot>))